modules/ocf/files/packages/cups/raster-filter

```diff
@@ -10,34 +10,52 @@ input=$(mktemp --suffix='.pdf')
 # Feed filter input into temporary file
 cat - > "$input"
 
+# If it's empty, check the 6th command line argument for the file
+if [ ! -s "$input" ]; then
+    cp "$6" "$input"
+fi
+
 # If the file is not a PDF, then just pass it thorugh
 mimetype="$(file -bi "$input")"
 
+echo "${mimetype%%;*}" > /tmp/mimetype
 if [ "${mimetype%%;*}" != "application/pdf" ]; then
-    cat "$input"
-    exit
-fi
-
-# Don't rasterize if size >1M
-size=$(du "$input" | cut -f1)
-[ "$size" -lt 1024 ] || exec cat "$input"
-
-# Or if size > 20 pages
-len=$(pdftk "$input" dump_data | awk '/NumberOfPages/{print $2}')
-[ -z "$len" -o "$len" -gt 20 ] && exec cat "$input"
-
-# Rasterizing PDF...
-output=$(mktemp --suffix='.pdf')
-error=$(mktemp)
-
-convert -density 250 "$input" "$output" 2> "$error"
-
-# Check exit status of convert
-if [ "$?" -ne 0 ]; then
-    cat "$error" | "/usr/local/bin/convert_failure"
-    cat "$input"
-    exit
+    exec cat "$input"
 fi
 
-# Done rasterizing
-cat "$output"
+# Returns a filename of a PDF to be processed further
+make_pdf() {
```

Comment:
Why is this a function if it's just getting called once immediately after being defined?